app/src/org/commcare/preferences/HiddenPreferences.java

```diff
@@ -320,4 +325,30 @@ public static boolean shouldShowUnsentFormsWhenZero() {
         SharedPreferences properties = CommCareApplication.instance().getCurrentApp().getAppPreferences();
         return properties.getString(SHOW_UNSENT_FORMS_WHEN_ZERO, PrefValues.NO).equals(PrefValues.YES);
     }
+
+    public static boolean preUpdateSyncNeeded() {
+        SharedPreferences properties = CommCareApplication.instance().getCurrentApp().getAppPreferences();
+        return properties.getString(PRE_UPDATE_SYNC_NEEDED, PrefValues.NO).equals(PrefValues.YES);
+    }
+
+    public static void setReleasedOnTimeForOngoingAppDownload(AndroidCommCarePlatform platform, long releasedOnTime) {
```

Comment:
There are a few places in the app where a pending update can be cleared, but I didn't see any way for this particular setter to be cleared out. Is there a situation where this gets zero'd because an update is abandoned?

Reply:
Currently there is not, since any new update check overrides the previous written value for this pref. But agree it makes sense to still clean this to protect against any incorrect usage in future.